### PR TITLE
Dialog

### PR DESCRIPTION
**Pull request by jcesarmobile**

### CODE REVIEW

**Comment by dwieeb** [comment]
@adamdbradley Needs PWA element for web side

File: dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            call.reject("Please provide a title or message for the alert");
+            return;
+        }
+
+        if (c.isFinishing()) {
+            call.reject("App is finishing");
+            return;
+        }
+
+        Dialog.alert(c, message, title, buttonTitle, (value, didCancel, inputValue) -> call.resolve());
```
Comment (carlpoole) [inline dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java:30]: rename `c` to `activity`, more descriptive

```suggestion
        final AppCompatActivity activity = this.getActivity();
        final String title = call.getString("title");
        final String message = call.getString("message");
        final String buttonTitle = call.getString("buttonTitle", "OK");

        if (title == null || message == null) {
            call.reject("Please provide a title or message for the alert");
            return;
        }

        if (activity.isFinishing()) {
            call.reject("App is finishing");
            return;
        }

        Dialog.alert(activity, message, title, buttonTitle, (value, didCancel, inputValue) -> call.resolve());
```

File: dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            return;
+        }
+
+        if (c.isFinishing()) {
+            call.reject("App is finishing");
+            return;
+        }
+
+        Dialog.confirm(
+            c,
```
Comment (carlpoole) [inline dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java:52]: rename `c` to `activity`, more descriptive

```suggestion
        final AppCompatActivity activity = this.getActivity();
        final String title = call.getString("title");
        final String message = call.getString("message");
        final String okButtonTitle = call.getString("okButtonTitle", "OK");
        final String cancelButtonTitle = call.getString("cancelButtonTitle", "Cancel");

        if (title == null || message == null) {
            call.reject("Please provide a title or message for the alert");
            return;
        }

        if (activity.isFinishing()) {
            call.reject("App is finishing");
            return;
        }

        Dialog.confirm(
            activity,
```

File: dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            return;
+        }
+
+        if (c.isFinishing()) {
+            call.reject("App is finishing");
+            return;
+        }
+
+        Dialog.prompt(
+            c,
```
Comment (carlpoole) [inline dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java:86]: rename `c` to `activity`, more descriptive

```suggestion
        final AppCompatActivity activity = this.getActivity();
        final String title = call.getString("title");
        final String message = call.getString("message");
        final String okButtonTitle = call.getString("okButtonTitle", "OK");
        final String cancelButtonTitle = call.getString("cancelButtonTitle", "Cancel");
        final String inputPlaceholder = call.getString("inputPlaceholder", "");
        final String inputText = call.getString("inputText", "");

        if (title == null || message == null) {
            call.reject("Please provide a title or message for the alert");
            return;
        }

        if (activity.isFinishing()) {
            call.reject("App is finishing");
            return;
        }

        Dialog.prompt(
            activity,
```

**Review by carlpoole — CHANGES_REQUESTED** [review]

File: dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java (hunk 3 above)
Comment (carlpoole) [reply dialog/android/src/main/java/com/capacitorjs/plugins/dialog/DialogPlugin.java:86]: Minor thing. Looks good!

File: dialog/src/definitions.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+   * Prepopulated text.
+   *
+   * @since 1.0.0
+   */
+  inputText?: string;
+}
+
+export interface ConfirmOptions {
+  /**
+   * Title of de dialog.
```
Comment (dwieeb) [inline dialog/src/definitions.ts:79]:
```suggestion
   * Title of the dialog.
```

File: dialog/src/definitions.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+   *
+   * @default "OK"
+   * @since 1.0.0
+   */
+  buttonTitle?: string;
+}
+
+export interface PromptOptions {
+  /**
+   * Title of de dialog.
```
Comment (dwieeb) [inline dialog/src/definitions.ts:33]:
```suggestion
   * Title of the dialog.
```

File: dialog/src/definitions.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
+declare module '@capacitor/core' {
+  interface PluginRegistry {
+    Dialog: DialogPlugin;
+  }
+}
+
+export interface AlertOptions {
+  /**
+   * Title of de dialog.
```
Comment (dwieeb) [inline dialog/src/definitions.ts:9]:
```suggestion
   * Title of the dialog.
```

**Review by ikeith — APPROVED** [review]

**Review by dwieeb — APPROVED** [review]

**Review by carlpoole — APPROVED** [review]

**Pull request merged**